src/MetaMorpho.sol

```diff
@@ -345,51 +345,52 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
     /// `supplied`).
     /// @dev The allocator can withdraw from any market, even if it's not in the withdraw queue, as long as the loan
     /// token of the market is the same as the vault's asset.
-    function reallocate(MarketAllocation[] calldata withdrawn, MarketAllocation[] calldata supplied)
-        external
-        onlyAllocatorRole
-    {
+    function reallocate(MarketAllocation[] calldata allocations) external onlyAllocatorRole {
+        uint256 totalSupplied;
         uint256 totalWithdrawn;
-        for (uint256 i; i < withdrawn.length; ++i) {
-            MarketAllocation memory allocation = withdrawn[i];
+        for (uint256 i; i < allocations.length; ++i) {
+            MarketAllocation memory allocation = allocations[i];
             Id id = allocation.marketParams.id();
 
             if (allocation.marketParams.loanToken != asset()) revert ErrorsLib.InconsistentAsset(id);
```

Comment:
This check is now done for both the supply and the withdraw case, but it is redundant in the supply case because we already check the cap (and a market with cap > 0 cannot have an inconsistent asset)